### PR TITLE
ADEN-8979 Add waitForViewed to Stickiness.

### PR DESCRIPTION
**Pull request by jacekbj**
Fix types for ad-bidders.

### CODE REVIEW

File: src/ad-services/bill-the-lizard/index.ts [hunk 1]
```diff
@@ -73,7 +73,7 @@ function buildUrl(host, endpoint, query) {
  * @returns {Promise}
  */
 function httpRequest(host, endpoint, queryParameters = {}, timeout = 0, callId) {
-	const request = new window.XMLHttpRequest();
+	const request = new (window as any).XMLHttpRequest() as XMLHttpRequest;
```
Comment (Bielik20) [inline src/ad-services/bill-the-lizard/index.ts]: How about `const request = new XMLHttpRequest();`?

File: src/ad-products/templates/uap/themes/hivi/stickiness.ts [hunk 2]
```diff
@@ -109,10 +113,12 @@ export class Stickiness extends EventEmitter {
 		this.applyStickiness();
 		this.logger('waiting for viewability and custom condition');
 
-		await Promise.all([
-			this.adSlot.viewed,
-			isFunction(this.customWhen) ? this.customWhen() : this.customWhen,
-		]);
+		const conditions = [isFunction(this.customWhen) ? this.customWhen() : this.customWhen];
```
Comment (Bielik20) [inline src/ad-products/templates/uap/themes/hivi/stickiness.ts]: How about doing it in constructor, like this:

```ts
constructor(customWhen: var | func = Promise.resolve()) {
  this.customWhen = isFunction(customWhen) ? customWhen : () => customWhen;
}
```

This way we would avoid those checks within the code, and we could be more precise with `CustomWhen` type definition.

File: src/ad-bidders/wrappers/apstag.ts [hunk 3]
```diff
@@ -14,22 +14,15 @@ export class Apstag {
...
-	 * @private
-	 */
-	renderImpEndCallbacks = [];
-
-	/**
-	 * @private
-	 * @type {boolean}
-	 */
-	renderImpHookPresent = false;
+	private renderImpEndCallbacks = [];
```
Comment (Bielik20) [inline src/ad-bidders/wrappers/apstag.ts]: 💎

File: src/ad-bidders/prebid/price-helper.ts [hunk 4]
```diff
@@ -35,45 +33,34 @@ function roundCpm(cpm, maxCpm) {
...
- * @param {string} slotName
- * @return {object}
- */
-export function getPrebidBestPrice(slotName) {
-	const bestPrices = {};
+export function getPrebidBestPrice(slotName: string): Dictionary<string> {
+	const bestPrices: Dictionary<number> = {};
 
 	if (window.pbjs && window.pbjs.getBidResponsesForAdUnitCode) {
 		const slotBids = window.pbjs.getBidResponsesForAdUnitCode(slotName).bids || [];
```
Comment (Bielik20) [inline src/ad-bidders/prebid/price-helper.ts]: type

File: src/ad-bidders/prebid/prebid-helper.ts [hunk 5]
```diff
@@ -26,7 +28,7 @@ function isSlotApplicable(code, lazyLoad) {
 	return !(isSlotDisabled || isSlotLazyIgnored);
 }
 
-export function setupAdUnits(lazyLoad = 'off') {
+export function setupAdUnits(lazyLoad = 'off'): AdUnitConfig[] {
 	const adUnits = [];
```
Comment (Bielik20) [inline src/ad-bidders/prebid/prebid-helper.ts]: type

File: src/ad-bidders/prebid/index.ts [hunk 6]
```diff
@@ -154,52 +174,46 @@ export class Prebid extends BaseBidder {
...
-	getBestPrice(slotName) {
+	getBestPrice(slotName: string): Dictionary<string> {
 		const slotAlias = this.getSlotAlias(slotName);
 
 		return getPrebidBestPrice(slotAlias);
 	}
 
-	getTargetingKeys(slotName) {
+	getTargetingKeys(slotName: string): string[] {
 		const allTargetingKeys = Object.keys(context.get(`slots.${slotName}.targeting`) || {});
```
Comment (Bielik20) [inline src/ad-bidders/prebid/index.ts]: type

File: src/ad-bidders/prebid/index.ts [hunk 7]
```diff
@@ -154,52 +174,46 @@ export class Prebid extends BaseBidder {
...
 			window.pbjs.removeAdUnit(adUnit.code);
 		});
 	}
 
-	/**
-	 * @inheritDoc
-	 */
-	getBestPrice(slotName) {
+	getBestPrice(slotName: string): Dictionary<string> {
 		const slotAlias = this.getSlotAlias(slotName);
```
Comment (Bielik20) [inline src/ad-bidders/prebid/index.ts]: type

File: src/ad-bidders/prebid/index.ts [hunk 8]
```diff
@@ -138,14 +158,14 @@ export class Prebid extends BaseBidder {
...
-	lazyCall(bidsBackHandler) {
+	lazyCall(bidsBackHandler: (...args: any[]) => void): void {
 		if (this.lazyLoaded) {
 			return;
 		}
 
 		this.lazyLoaded = true;
 
-		const adUnitsLazy = setupAdUnits(this.bidderConfig, 'post');
+		const adUnitsLazy = setupAdUnits('post');
```
Comment (Bielik20) [inline src/ad-bidders/prebid/index.ts]: type

File: src/ad-services/bill-the-lizard/index.ts (hunk 1 above)
Comment (jacekbj) [reply src/ad-services/bill-the-lizard/index.ts]: 🤦‍♂

File: src/ad-products/templates/uap/themes/hivi/stickiness.ts (hunk 2 above)
Comment (jacekbj) [reply src/ad-products/templates/uap/themes/hivi/stickiness.ts]: As discussed on Slack, this is a good topic for refactoring. However, in order not to increase the scope of testing, I decided to leave it for now.

File: src/ad-bidders/wrappers/apstag.ts (hunk 3 above)
Comment (jacekbj) [reply src/ad-bidders/wrappers/apstag.ts]: 🙇

**Pull request merged**